client_src/js/printerManager/printer-constructor.js

```diff
@@ -0,0 +1,323 @@
+import Validate from "../lib/functions/validate";
+import UI from "../lib/functions/ui";
+import OctoFarmClient from "../lib/octofarm_client.js";
+
+let newPrintersIndex = 0;
+
+const removeLine = function (element) {
+  element.remove();
+};
+class Printer {
+  constructor(printerURL, camURL, apikey, group, name) {
+    this.settingsAppearance = {
+      color: "default",
+      colorTransparent: false,
+      defaultLanguage: "_default",
+      name,
+      showFahrenheitAlso: false
+    };
+    this.printerURL = printerURL;
+    this.camURL = camURL;
+    this.apikey = apikey;
+    this.group = group;
+  }
+}
+export class PrintersManagement {
+  constructor(printerURL, camURL, apikey, group, name) {
+    this.printer = new Printer(printerURL, camURL, apikey, group, name);
+  }
+
+  static addPrinter(newPrinter) {
+    // Insert Blank Row at top of printer list
+    if (
+      document.getElementById("printerNewTable").classList.contains("d-none")
+    ) {
+      document.getElementById("printerNewTable").classList.remove("d-none");
+    }
+
+    if (typeof newPrinter !== "undefined") {
+      document.getElementById("printerNewList").insertAdjacentHTML(
```

Comment:
Is this a UI manager or a data manager?

Reply:
It's an old god awful amalgamation of both, left a note to clean up that file as it's got stuff tagged on from old usages.